src/nft/hooks/useBag.ts

```diff
@@ -1,22 +1,61 @@
-import { BagItem } from 'nft/types'
+import { v4 as uuidv4 } from 'uuid'
 import create from 'zustand'
 import { devtools } from 'zustand/middleware'
 
-interface BagState {
+import { BagItem, BagItemStatus, BagStatus, UpdatedGenieAsset } from '../types'
+
+type BagState = {
   itemsInBag: BagItem[]
+  addAssetToBag: (asset: UpdatedGenieAsset) => void
+  removeAssetFromBag: (asset: UpdatedGenieAsset) => void
+  isLocked: boolean
+  setLocked: (isLocked: boolean) => void
   bagExpanded: boolean
   toggleBag: () => void
 }
 
 export const useBag = create<BagState>()(
   devtools(
-    (set) => ({
+    (set, get) => ({
       bagExpanded: false,
-      itemsInBag: [],
       toggleBag: () =>
         set(({ bagExpanded }) => ({
           bagExpanded: !bagExpanded,
         })),
+      isLocked: false,
+      setLocked: (_isLocked) =>
+        set(() => ({
+          isLocked: _isLocked,
+        })),
+      itemsInBag: [],
+      addAssetToBag: (asset) =>
+        set(({ itemsInBag }) => {
+          if (get().isLocked) return { itemsInBag }
+          const assetWithId = { asset: { id: uuidv4(), ...asset }, status: BagItemStatus.ADDED_TO_BAG }
+          if (itemsInBag.length === 0)
+            return {
+              itemsInBag: [assetWithId],
+              bagStatus: BagStatus.ADDING_TO_BAG,
+            }
+          else
+            return {
+              itemsInBag: [...itemsInBag, assetWithId],
+              bagStatus: BagStatus.ADDING_TO_BAG,
+            }
+        }),
+      removeAssetFromBag: (asset) => {
```

Comment:
can you only add the parts of useBag that are actively being used by Asset? Don't want to accidentally bring in broken code that can't currently be tested

Reply:
this is all used with the exception of setLocked.  It didn't make sense to leave it out since isLocked is used.  I think this is ok for now